src/crypto-providers/util.ts

```diff
@@ -357,20 +357,15 @@ const _packBootStrapAddress = (
   return {
     address,
     addressType: getAddressType(address),
-    paymentPath: hwsData.path,
+    paymentPath: paymentSigningFile.path,
   }
 }
 
 const _packBaseAddress = (
-  hwsData: HwSigningData[], network: Network,
+  paymentSigningFile: HwSigningData, stakeSigningFile: HwSigningData, network: Network,
 ): _AddressParameters | null => {
-  const isStakingPath = (path: number[]) => classifyPath(path) === PathTypes.PATH_WALLET_STAKING_KEY
-  const stakePathFile = hwsData.find(({ path }) => isStakingPath(path))
-  const paymentPathFile = hwsData.find(({ path }) => !isStakingPath(path))
-  if (!stakePathFile || !paymentPathFile) return null
-
-  const { pubKey: stakePubKey } = destructXPubKeyCborHex(stakePathFile.cborXPubKeyHex)
-  const { pubKey: paymentPubKey } = destructXPubKeyCborHex(paymentPathFile.cborXPubKeyHex)
+  const { pubKey: stakePubKey } = destructXPubKeyCborHex(stakeSigningFile.cborXPubKeyHex)
```

Comment:
What about replacing `destruct` with separate / split / ... or even better `extractXPubKeyFromCborHex`? Seeing `destruct` irritates me every single time I see it.

Reply:
Renamed to split, there is a place in code where chain code is used well.